src/Games/NexusMods.Games.FOMOD/CoreDelegates/UiDelegate.cs

```diff
@@ -69,74 +86,187 @@ public void EndDialog()
         _logger.LogError("Reporting an error: {Title}: {Message}\n{Details}", title, message, details);
     }
 
-    public void UpdateState(InstallerStep[] installSteps, int currentStepId)
+    public void UpdateState(FomodInstaller.Interface.ui.InstallerStep[] installSteps, int currentStepId)
     {
-        var currentStep = installSteps[currentStepId];
-        var choices = ToChoices(currentStep.optionalFileGroups).ToArray();
+        // NOTE(erri120): The FOMOD library we're using can send us stupid inputs.
+        if (currentStepId < 0 || currentStepId >= installSteps.Length) return;
+
+        // NOTE(erri120): This fuckery is explained further below when we call _selectOptions()
+        if (Interlocked.Read(ref _taskFuckeryState) == WaitingForCallback)
```

Comment:
You can leave the comments, but I think the `_taskFuckeryState` variable should have a more expressive name, something like `_taskWaitingState`